src/TuxState.js

```diff
@@ -0,0 +1,650 @@
+  //@param propsToDivide OBJECT: required object argument with deepest keys being numbers to divide the current state by
+  //@param callback FUNCTION: optional callback argument that will be executed once setState is completed and the component is re-rendered
+  divideState: function (propsToDivide, callback) {
+    //throw error if propsToDivide is undefined or not an object
+    invariantArgCheck(propsToDivide);
+    //make a shallow copy of the current state
+    var currentState = assign({}, this.state);
+    //build new state object
+    var newState = buildNewState(currentState, propsToDivide, function(newState, key, currentStateAtKey, newPropsAtKey) {
+      //throw error if newPropsAtKey is not a number
+      invariantNumberCheck(newPropsAtKey);
+      //perform division on the corresponding deep keys
+      newState[key] = currentStateAtKey / newPropsAtKey;
+    });
+    //set state with the updated values and a callback (if provided), triggering a re-render
+    this.setState(newState, callback);
+  },
+
+  //omitState FUNCTION: removes the deepest keys in the passed in object from the corresponding deepest keys in the current state, or throws an error if keys don't match
+  //@param propsToOmit OBJECT: required object argument with deepest keys being booleans
+  //@param callback FUNCTION: optional callback that will be executed once replaceState is completed and the component is re-rendered
+  omitState: function (propsToOmit, callback) {
+    //throw error if propsToOmit is undefined or not an object
+    invariantArgCheck(propsToOmit);
+    //make a shallow copy of the current state
+    var currentState = assign({}, this.state);
+    //build new state object
+    var newState = buildNewState(currentState, propsToOmit, function(newState, key, currentStateAtKey, newPropsAtKey) {
+      //remove the deepest key
+      delete newState[key];
+    });
+    //replace current state with the updated values and a callback (if provided), triggering a re-render
+    this.replaceState(newState, callback);
+  },
+
+  //extendState FUNCTION: adds new keys from the passed in object to the current state and overrides pre-existing keys, throws error if no outer keys in passed in object match outer keys in current state
+  //@param propsToExtend OBJECT: required object argument with at least one outer key matching an outer key in the current state, and any additional keys being keys to add to the current state
+  //@param callback FUNCTION: optional callback that will be executed once setState is completed and the component is re-rendered
+  extendState: function (propsToExtend, callback) {
+    //throw error if propsToExtend is undefined or not an object
+    invariantArgCheck(propsToExtend);
+    //make a shallow copy of the current state
+    var currentState = assign({}, this.state);
```

Comment:
I don't think we need to invoke `assign` here because we are already making a shallow copy of state at the first line in `buildNewState`.  Below compare propsToExtend with `this.state` instead.